pkg/zcapld/verify.go

```diff
@@ -193,7 +194,14 @@ func (v *Verifier) verifyCapabilityChain(
 	}
 
 	// 4.2. Ensure that the caveats are met on the root capability.
-	// TODO verify caveats
+	for i := range capability.Caveats {
+		if capability.Caveats[i].Type == CaveatTypeExpiry {
```

Comment:
This is the reason why I think the mechanism should be generalized: users should be able to inject their own caveat validators.

Reply:
got it